iron-list.html

```diff
@@ -486,19 +490,85 @@
         }
       }
 
-     // update the position of the items
+      // update the position of the items
       this._positionItems();
 
       // set the scroller size
       this._updateScrollerSize();
+
+      if (this._increasePoolIfNeeded()) {
+        // set models to the new items
+        this.async(this._update);

```

Comment:
Is there a reason to make this async, other than to ensure the call stack doesn't nest?  It shouldn't happen more than once or twice in a turn, so I don't see this being a problem...